core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -172,44 +158,31 @@ public void addKeyStore(byte[] keystoreContent, String filename, String password
         addKeystore(filename, keystoreContent, password, provider, type, privateKeyPass);
     }
 
-    private void addKeystore(String filename, byte[] keystoreContent, String password, String provider,
-                             String type, String privateKeyPass) {
+    private void addKeystore(String filename, byte[] keystoreContent, String password, String provider, String type,
+                             String privateKeyPass) {
+
+        if (keyStoreRegistryPersistenceManager.isKeyStoreExists(filename)) {
+            throw new SecurityException("Key store " + filename + " already available");
+        }
 
-        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + filename;
-        boolean isTenantPrimaryKeyStore = false;
         try (InputStream inputStream = new ByteArrayInputStream(keystoreContent)) {
-            if (registry.resourceExists(path)) {
-                throw new SecurityException("Key store " + filename + " already available");
-            }
-            if (tenantId != MultitenantConstants.SUPER_TENANT_ID &&
-                    !registry.resourceExists(RegistryResources.SecurityManagement.KEY_STORES)) {
-                isTenantPrimaryKeyStore = true;
-            }
+
             KeyStore keyStore = KeystoreUtils.getKeystoreInstance(type);
             keyStore.load(inputStream, password.toCharArray());
 
-            Resource resource = registry.newResource();
-            resource.addProperty(RegistryResources.SecurityManagement.PROP_PASSWORD, encryptPassword(password));
-            resource.addProperty(RegistryResources.SecurityManagement.PROP_PROVIDER, provider);
-            resource.addProperty(RegistryResources.SecurityManagement.PROP_TYPE, type);
-
             String pvtKeyAlias = KeyStoreUtil.getPrivateKeyAlias(keyStore);
             if (StringUtils.isNotBlank(pvtKeyAlias)) {
-                resource.addProperty(RegistryResources.SecurityManagement.PROP_PRIVATE_KEY_ALIAS, pvtKeyAlias);
                 if (StringUtils.isNotBlank(privateKeyPass)) {
-                    resource.addProperty(RegistryResources.SecurityManagement.PROP_PRIVATE_KEY_PASS,
-                            encryptPassword(privateKeyPass));
```

Comment:
We can encrypt the private key pass here

Reply:
Fixed in the latest changes